packages/compiler/src/process/parser/yaml/index.ts

```diff
@@ -31,18 +31,32 @@ export type Preprocess<T = Trigger | Task> = {
   replace(step: T): Promise<T>
 }
 
+export type YAMLStruct = {
+  workflows: {
+    [key: string]: (Trigger | Filter | Task)[]
```

Comment:
doesn't it make more sense to just put the filepath to each workflow reather that the whole workflows?
Do you want to implement both system?
```suggestion
    [key: string]: string[]
```

On one side, implementing both system gives the best flexibility but we'll need to again implement some kind of content detection (if data is string, then read file, else data is workflow) and I would love to not do the same confusing grammar that the current compiler has.

Reply:
I will be ok to have a explicit grammar like you suggested on the doc PR:
```
$(file: foo/bar.yml)
```
You told me that YAML implemented some kind of class system? How does it look like and work?